src/function_classes/retrieval.py

```diff
@@ -1,22 +1,31 @@
 import torch
 import torch.distributions as D
 
+from torch import Tensor
+
 from core import FunctionClass
+from utils import CombinedDistribution
 
 class Retrieval(FunctionClass):
 
     def _init_param_dist(self) -> D.Distribution:
-        return D.Categorical(torch.Tensor([1]))
-    
-    def evaluate(self, x_batch: torch.Tensor, *params: torch.Tensor) -> torch.Tensor:
-        assert x_batch.shape[0] == self.batch_size and x_batch.shape[2] == self.x_dim
-        assert x_batch.shape[1] % 2 == 1
```

Comment:
is it still worth having an evaluate method for this function class even though we're overriding the `__next__` method? (is evaluate only useful for `__next__` or does it need to be used in other places as well?)

Reply:
This is a good point -- since "evaluation" involves also mutating our x_batch, it might be a good idea to have a thin `evaluate` definition that raises an informative error when called